packages/flutter_tools/lib/src/android/android_device.dart

```diff
@@ -134,6 +135,42 @@ class AndroidDevice extends Device {
     return _isLocalEmulator;
   }
 
+  /// If the device is an emulator, returns the ID shown in the output of
+  /// `flutter emulators` that matches this device. Fetching this name may require
+  /// connecting to the device and if an error occurs null will be returned.
+  /// If this devices is not an emulator, null will be returned.
+  @override
+  Future<String> get emulatorId async {
+    // Emulators always have IDs in the format emulator-(port) where port is the
+    // Android Console port number.
+    final RegExp emulatorPortRegex = RegExp(r'emulator-(\d+)');
+    if (!(await isLocalEmulator))
+      return null;
+
+    try {
+      final Match portMatch = emulatorPortRegex.firstMatch(id);
+      if (portMatch == null || portMatch.groupCount < 1) {
+        return null;
+      }
+
+      const String host = 'localhost';
+      final int port = int.parse(portMatch.group(1));
+      printTrace('Connecting to $host:$port to get avd name');
+      final AndroidConsole console = await AndroidConsole.connect('localhost', port);
+      try {
+        return await console.getAvdName();
+      } finally {
+        console.destroy();
+      }
+    }
+    catch (e) {
```

Comment:
Nit: move catch clause to same line as closing `}`

Reply:
Done (though this code has changed a little now too).